### PR TITLE
Bubble up LRO errors when request fail during polling

### PR DESCRIPTION
**Pull request by shahabhijeet**
This fixes issue #3396 
Currently errors that are returned in LRO responses are not passed on to clients. This PR fixes that.
Added tests
Cleaned up build warnings

### CODE REVIEW

File: src/SdkCommon/ClientRuntime.Azure/ClientRuntime.Azure/AzureClientExtensions.cs [hunk 1]
```diff
@@ -356,6 +356,8 @@ public static class AzureClientExtensions
             }
             return true;
         }
+   
+/*
```
Comment (hovsepm) [inline src/SdkCommon/ClientRuntime.Azure/ClientRuntime.Azure/AzureClientExtensions.cs]: please remove commented code.

File: src/SdkCommon/ClientRuntime.Azure/ClientRuntime.Azure/AzureLROOperationExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+                    };
+
+                    pollingState.CloudException = new CloudException(errorMessage)
+                    {
+                        Body = asyncOperation.Error,
+                        Request = new HttpRequestMessageWrapper(pollingState.Request, null),
+                        Response = new HttpResponseMessageWrapper(pollingState.Response, responseContent)
+                    };
+                }
+                //}
```
Comment (hovsepm) [inline src/SdkCommon/ClientRuntime.Azure/ClientRuntime.Azure/AzureLROOperationExtensions.cs]: >//} [](start = 16, length = 3)

please remove this commented line as well

**Comment by hovsepm** [comment]
rest LGTM

File: src/SdkCommon/ClientRuntime.Azure/ClientRuntime.Azure/AzureClientExtensions.cs (hunk 1 above)
Comment (shahabhijeet) [inline src/SdkCommon/ClientRuntime.Azure/ClientRuntime.Azure/AzureClientExtensions.cs]: @hovsepm the plan was to keep this code for two reason:
1) makes it easy to code review the old logic and the new
2) Also for any unforeseen bugs, it's easy to test it against the old logic by uncommenting the code.
3) In next few weeks, once we publish this nuget package, the commented code will be removed.

**Pull request merged**